scipy/fftpack/logarithmic.py

```diff
@@ -0,0 +1,256 @@
+"""
+FFTLog - fftlog.py
+"""
+# Created by Dieter Werthmüller, January 2017
+from __future__ import division, print_function, absolute_import
+
+import atexit
+import numpy as np
+from . import _fftl
+from .basic import _asfarray
+
+__all__ = ['fftlog', 'fftlogargs']
+
+atexit.register(_fftl.destroy_fftl_x_cache)
+atexit.register(_fftl.destroy_fftl_w_cache)
+del atexit
+
+
+def fftlog(x, dlogr, mu='sine', q=0.0, kr=1.0, rk=1.0):
+    """Fourier transform of a logarithmically spaced periodic sequence.
+
+    Fast Fourier transform of a real, discrete periodic sequence of
+    logarithmically spaced points.
+
+    `fftlog` computes a discrete version of the Fourier sine (if mu = 1/2) or
+    cosine (if mu = -1/2) transform
+
+    .. math::
+
+        \tilde{A} = \sqrt{2/\pi} \int_0^\infty A(r) \sin(kr) dr,
+
+        \tilde{A} = \sqrt{2/\pi} \int_0^\infty A(r) \cos(kr) dr
+
+
+    by making the substitutions
+
+    .. math::
+
+        A(r) = a(r) r^{\mu-1/2},
+
+        \tilde{A}(k) = \tilde{a}(k) k^{-\mu -1/2}
+
+    and applying a biased Hankel transform to a(r).
+
+    Parameters
+    ----------
+    x : array_like, real-valued
+        Array A(r) to transform: a(j) is A(r_j) at r_j = r_c exp[(j-jc) dlnr],
+        where jc = (n+1)/2 = central index of array.
+
+    dlogr : float, optional
+        Separation between input-points (log10); may be positive or negative.
+        Default is 0.01.
+
+    mu : string, optional; {'sine', 'cosine'}
+        Index of J_mu in Hankel transform: 0.5 for a sine transform and -0.5
+        for a cosine transform. Default is 'sine' (0.5).
+
+    q : float, optional
+        Exponent of power law bias; q may be any real number, positive or
+        negative.  If in doubt, use q = 0, for which case the Hankel transform
+        is orthogonal, i.e. self-inverse, provided also that, for n even, kr is
+        low-ringing.  Non-zero q may yield better approximations to the
+        continuous Hankel transform for some functions. Default is 0
+        (unbiased).
+
+    kr : float, optional
+        k_c r_c where c is central point of array
+        = k_j r_(n+1-j) = k_(n+1-j) r_j .
+        Normally one would choose kr to be about 1 (default) (or 2, or pi, to
+        taste). Default is 1.
+
+    rk : float, optional
+        r_c/k_c = r_j/k_j (a constant, the same constant for any j); rk is not
+        (necessarily) the same quantity as kr.  rk is used only to multiply the
+        output array by sqrt(rk)^dir, so if you want to do the normalization
+        later, or you don't care about the normalization, you can set rk = 1.
+        Default is 1.
+
+    n : int, optional
+        Defines the length of the Fourier transform.  If `n` is not specified
+        (the default) then ``n = x.shape[axis]``.  If ``n < x.shape[axis]``,
+        `x` is truncated, if ``n > x.shape[axis]``, `x` is zero-padded.
+
+    axis : int, optional
```

Comment:
this parameter as well as ``overwrite_x`` is not in the signature of this function